lib/index.js

```diff
@@ -397,18 +397,24 @@ function initDB(peer, config) {
    * Find which sigkeys are authorized to sign this msg given the account.
    *
    * @private
-   * @param {Tangle | null} accountTangle
+   * @param {DBTangle | null} accountTangle
+   * @param {Msg['metadata']['accountTips']} accountTipsInMsg
    * @returns {Set<string>}
    */
-  function getSigkeysInAccount(accountTangle) {
+  function getSigkeysInAccount(accountTangle, accountTipsInMsg) {
     console.log('accountTangle', accountTangle?.debug())
     // TODO: add test with more than 1 msg in the account tangle
     //getSigkeysInAccount needs a new argument: accountTips of type Msg['metadata']['accountTips']
     //use tangle.slice()
     const sigkeys = new Set()
     if (!accountTangle) return sigkeys
-    // TODO: prune the accountTangle beyond msg.metadata.accountTips
-    for (const msgID of accountTangle.topoSort()) {
+
+    const prunedTangle = accountTangle
+      .slice(undefined, accountTipsInMsg ?? undefined)
+      .map((msg) => MsgV4.getMsgID(msg))
+    console.log('prunedTangle', prunedTangle)
+
+    for (const msgID of prunedTangle) {
```

Comment:
This can be a for-loop of `msg` not `msgID`. Notice how just below we do `const msg = get(msgID)` so there's no point converting from msg to msgID (two lines above) just to convert back from msgID to msg (one line below). For performance.

Reply:
ah lol smart